File: Leger/PolyPaint/app/src/main/java/com/polypaint/polypaint/Model/ClassShape.kt
package com.polypaint.polypaint.Model


class ClassShape (id: String, type: Int, name: String, shapeStyle: ShapeStyle, links: ArrayList<String?> = ArrayList(), linksFrom: ArrayList<String?> = ArrayList(), var attributes: ArrayList<String?> = ArrayList(), var methods: ArrayList<String?> = ArrayList()):BasicShape(id, type, name, shapeStyle, links, linksFrom){

    fun copyClass(id: String = this.id, type: Int = this.type, name: String = this.name, shapeStyle: ShapeStyle = this.shapeStyle.copy(), linksTo: ArrayList<String?> = ArrayList<String?>(), linksFrom: ArrayList<String?> = ArrayList<String?>(), attributes: ArrayList<String?> = this.attributes.clone() as ArrayList<String?>, methods: ArrayList<String?> = this.methods.clone() as ArrayList<String?>) = ClassShape(id, type, name, shapeStyle, linksTo, linksFrom, attributes, methods)

}

File: Lourd/PolyPaint/PolyPaint/Templates/ClassShape.cs
using PolyPaint.CustomInk;
using System.Collections.Generic;

namespace PolyPaint.Templates
{
    public class ClassShape : BasicShape
    {
        public List<string> attributes { get; set; }
        public List<string> methods { get; set; }

        public ClassShape()
        {
        }

        public ClassShape(string id, int type, string name, ShapeStyle shapeStyle, List<string> linksTo, List<string> linksFrom, List<string> attributes, List<string> methods)
        {
            this.id = id;
            this.type = type;
            this.name = name;
            this.shapeStyle = shapeStyle.Clone();
            this.shapeStyle.coordinates.x *= CustomStroke.WIDTH;
            this.shapeStyle.coordinates.y *= CustomStroke.HEIGHT;
            this.shapeStyle.width *= CustomStroke.WIDTH;
            this.shapeStyle.height *= CustomStroke.HEIGHT;
            this.linksTo = linksTo;
            this.linksFrom = linksFrom;
            this.attributes = attributes;
            this.methods = methods;
        }

    }
}